webui/react/src/components/ResourcePoolCard.tsx

```diff
@@ -115,7 +114,9 @@ const ResourcePoolCard: React.FC<Props> = ({
   }, [pool]);
 
   const processedPool = useMemo(() => {
-    const newPool = clone(pool);
+    // had to add this in order to avoid adding ts-ignore at lines 121 and 122 due to incompatibility of types
+    // since this is sort of a translation of the API response object into UI, this should be fine.
+    const newPool = JSON.parse(JSON.stringify(structuredClone(pool)));
```

Comment:
looks like the original code type is incorrect. I think this should be fixed with correct types instead of implicitly ignore it

Reply:
i dont think the change would be big because it looks like this is the only place that return `any` around `structuredClone`, so i think this should be fixed in this pr